pennylane/_grad.py

```diff
@@ -21,55 +21,101 @@
 from autograd.numpy.numpy_boxes import ArrayBox
 from autograd.extend import vspace
 from autograd.wrap_util import unary_to_nary
+from pennylane.compiler import compiler
 
 make_vjp = unary_to_nary(_make_vjp)
 
 
 class grad:
-    """Returns the gradient as a callable function of (functions of) QNodes.
+    """A :func:`~.qjit` compatible gradient transformation that returns the gradient
+    as a callable function of (functions of) QNodes.
 
-    By default, gradients are computed for arguments which contain the property
-    ``requires_grad=True``. Alternatively, the ``argnum`` keyword argument can
-    be specified to compute gradients for function arguments without this property,
+
+    By default, in interpreted mode, gradients are computed for arguments which contain
```

Comment:
yes, I was struggling about naming these two modes in docs as well. I supposed "interpreted", "greedy" and " (JIT) compilation" are now buzzwords and everyone has some clue about them. What do you think we should use instead? cc: @trbromley @josh146

Reply:
We can move this further down the docstring, but tbh I am not too worried, and wouldn't actually mind if users read this, got curious, and explored further.

Reply:
What is our objective with mentioning `qjit` in this function? I can think of the following justifications:
1. To help identify what is compatible with `qjit`/Catalyst.
2. To have `qjit` mentioned more throughout the docs, raising awareness.

For point 1, I think this would be better served with a single source of truth, e.g., a table within the `qjit` usage details that specifies what currently works (in the long run, should everything work?). It seems more challenging to maintain individual function docstrings.

However, I appreciate point 2 is also important. Would it work to have a simple Note/Info box above the "grad is intended to be used with the Autograd interface only." warning. The box can simply say "This function is compatible with `qjit()`, check out the quickstart guide for more details! ⚗️"? Having a standalone box should be easier to maintain.

Reply:
One datapoint from our core standup - those who are not familiar with just-in-time compilation may be dissuaded from reading these docstrings: "Oh this function is about JIT? Doesn't apply to me then!".

I think we should still mention `qjit()` here, I just think we need to word it carefully.

Reply:
@trbromley happy with any of your suggestions here. But would still like to include a :func:`~.qjit` example in this docstring.

Something to note is that as we deprecate autograd, this function will remain as a pure qjit-compatible gradient function

Reply:
Fixed #4812  (chatted offline with Tom)